tools/apigee-envoy-quickstart/extension/setup-external-access.sh

```diff
@@ -0,0 +1,223 @@
+    -f "$ENVOY_HOME"/asm/samples/gateways/istio-ingressgateway
+
+  kubectl rollout restart deployment -n $ASM_GATEWAYS_NAMESPACE
+
+  getIngressGatewayExternalIP;
+
+  counter=0;
+  while [[ "$EXTERNAL_IP" == 'null' ]] && [[ $counter -lt 10 ]]; do
+    printf "\n\nTesting the availability of EXTERNAL_IP for ingressgateway,  %s of 10\n" "$counter"
+    sleep 5
+    getIngressGatewayExternalIP;
+    counter=$((counter+1))
+  done
+
+}
+
+fetchConsumerKey() {
+  CONSUMER_KEY=$(curl -H "Authorization: ${TOKEN_TYPE} ${TOKEN}"  \
+    -H "Content-Type:application/json" \
+    "${MGMT_HOST}/v1/organizations/${APIGEE_ORG}/developers/test-envoy@google.com/apps/envoy-adapter-app-2" | \
+    jq '.credentials[0].consumerKey'); \
+    CONSUMER_KEY=$(echo "$CONSUMER_KEY"|cut -d '"' -f 2); export CONSUMER_KEY; \
+    printf "\n" && printf "\n"
+
+  if [[ -z "$CONSUMER_KEY" ]] || [[ "$CONSUMER_KEY" == 'null' ]]; then
+      echo "Failed in extracting the CONSUMER_KEY of the developer app attached to the httpbin product in Apigee org"
+      exit 1
+  fi
+}
+
+testExternalAccess() {
+  RESULT=1
+  OUTPUT=$(curl -i http://"$EXTERNAL_IP"/headers -H 'Host: httpbin.apigee.svc.cluster.local' \
+      -H "x-api-key: $CONSUMER_KEY" | grep HTTP)
+
+  printf "\n%s" "$OUTPUT"
+  if [[ "$OUTPUT" == *"200"* ]]; then
+      RESULT=0
+  fi
+  return $RESULT
+}
+
+echo 'Initializing...'
+init;
+
+echo 'Validating...'
+validate;
+
+ISTO_INGRESSGATEWAY_CNT=$(kubectl --context="${CLUSTER_CTX}" get svc -n "$ASM_GATEWAYS_NAMESPACE" -o json | \
+jq '.items|length')
+
+if [ "$ISTO_INGRESSGATEWAY_CNT" == 0 ]; then
+  echo "No istio ingressgateway found, installing..."
+  installIstioIngressGateway;
+fi
+
+echo "Getting ingressgateway external ipaddress..."
+getIngressGatewayExternalIP;
+
+if [[ -z "$EXTERNAL_IP" ]] || [[ "$EXTERNAL_IP" == 'null' ]]; then
+    echo "ExternalIP is not set on the istio ingressgateway, exiting"
+    exit 1
+fi
+
+kubectl --context="${CLUSTER_CTX}" get svc -n "$ASM_GATEWAYS_NAMESPACE"
+
+echo "Installing gateway for httpbin..."
+cat <<EOF > "${ENVOY_HOME}"/httpbin-gateway.yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+ name: httpbin-gateway
+spec:
+ selector:
+   istio: ingressgateway
+ servers:
+   - port:
+       number: 80
+       name: http
+       protocol: HTTP
+     hosts:
+       - "*"
+EOF
+
+kubectl apply -n $ASM_GATEWAYS_NAMESPACE \
+  -f "$ENVOY_HOME"/httpbin-gateway.yaml
+
+echo "Installing virtualservice for httpbin..."
+cat <<EOF > "${ENVOY_HOME}"/httpbin-virtual-service.yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: httpbin-ingress
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - httpbin-gateway
+  http:
+  - route:
+    - destination:
+        host: httpbin.apigee.svc.cluster.local
+        port:
+          number: 80
+EOF
+
+kubectl apply -n $ASM_GATEWAYS_NAMESPACE \
+  -f "$ENVOY_HOME"/httpbin-virtual-service.yaml
+
+fetchConsumerKey;
```

Comment:
consider writing it as:

```sh
CONSUMER_KEY=$(fetchConsumerKey)
```

where the `fetchConsumerKey` function writes the key to stdout. And the error to stderr.